x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/MvSort.java

```diff
@@ -147,10 +151,20 @@ public boolean foldable() {
     public EvalOperator.ExpressionEvaluator.Factory toEvaluator(
         Function<Expression, EvalOperator.ExpressionEvaluator.Factory> toEvaluator
     ) {
-        Expression nonNullOrder = order == null ? ASC : order;
-        boolean ordering = nonNullOrder.foldable() && ((BytesRef) nonNullOrder.fold()).utf8ToString().equalsIgnoreCase("DESC")
-            ? false
-            : true;
+        boolean ordering = true;
```

Comment:
One more nit: can we somehow minimise the duplications here? An idea (maybe there are others) might be to have a `Boolean ordering(Expression)` method returning `null` if invalid? Also, both the `InvalidArgumentException` c'tor and `Failure.fail()` take a string, maybe we can construct that one in a single place.

Reply:
Another idea would be to use `resolveType()` for this verification (see TopList).

Reply:
> Another idea would be to use `resolveType()` for this verification (see TopList).

I moved the validation of order from `toEvaluator` to `resolveType()`.